src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -56,12 +56,22 @@ private void ReturnReadBufferIfEmpty()
         }
 
         ~SslStreamInternal()
+        {
+            Close(disposing: false);
+        }
+
+        internal void Close(bool disposing)
         {
             if (_internalBuffer != null)
             {
                 ArrayPool<byte>.Shared.Return(_internalBuffer);
                 _internalBuffer = null;
```

Comment:
What happens if Close is called while there are still pending operations? Might we return a buffer to the pool while it's still in use?

Reply:
Yeah that could happen, when dispose is called on some thread, and the thread with readasync/writeasync is copying data between _internalBuffer to user buffer.

Reply:
And that's ok? From the other thread about disposing an in-use SslStream and expecting operations to be canceled, it sounds like that's a pattern people use.

Reply:
Ah, I see you addressed it in your updates. Thanks.